components/profile.js

```diff
@@ -103,15 +102,273 @@ SteamCommunity.prototype.editProfile = function(settings, callback) {
+									}
+									if (settings[i][type]["values"].hasOwnProperty("notes")) {
+										values["rgShowcaseConfig[8][0][notes]"] = settings[i][type]["values"]["notes"];
+									}
+								}
+
+								break;
+
+							case 'artwork':
+								values["profile_showcase[13]"] = 13;
+
+								if (settings[i][type].hasOwnProperty("values")) {
+									for (var n = 0; n < 4; n++) {
+										values["profile_showcase[13][" + n + "][publishedfileid]"] = settings[i][type]["values"][n] || "";
+									}
+								}
+								break;
+
+							case 'trade':
+								values["profile_showcase[4]"] = 4;
+
+								if (settings[i][type].hasOwnProperty("values")) {
+
+									if (settings[i][type]["values"].hasOwnProperty("notes")) {
+										values["rgShowcaseConfig[4][6][notes]"] = settings[i][type]["values"]["notes"];
+									}
+
+									if (settings[i][type]["values"].hasOwnProperty("items")) {
+										showcaseconfig.supplied = true;
+										showcaseconfig.numberofrequests = 6;
+										showcaseconfig.showcasetype = 4;
+										showcaseconfig.itemarray = settings[i][type]["values"]["items"];
+									}
+								}
+								break;
+
+							case 'items':
+								values["profile_showcase[3]"] = 3;
+
+								if (settings[i][type].hasOwnProperty("values") && settings[i][type]["values"].hasOwnProperty("items")) {
+									showcaseconfig.supplied = true;
+									showcaseconfig.numberofrequests = 10;
+									showcaseconfig.showcasetype = 3;
+									showcaseconfig.itemarray = settings[i][type]["values"]["items"];
+								}
+								break;
+
+							case 'game':
+								values["profile_showcase[6]"] = 6;
+
+								if (settings[i][type].hasOwnProperty("values")) {
+									values["rgShowcaseConfig[6][0][appid]"] = settings[i][type]["values"];
+								}
+								break;
+
+							case 'badge':
+								values["profile_showcase[5]"] = 5;
+
+								if (settings[i][type].hasOwnProperty("values")) {
+
+									if (settings[i][type]["values"].hasOwnProperty("style")) {
+										var styles = ["rare", "selected", null, "recent", "random"];
+										values["profile_showcase_style_5"] = styles.indexOf(settings[i][type]["values"]["style"]);
+									}
+
+									if (settings[i][type]["values"].hasOwnProperty("badges")) {
+										var types = ["badgeid", "appid", "border_color"];
+										for (var n = 0; n < 6; n++) {
+											for (var m = 0; m < 3; m++){
+												if (settings[i][type]["values"]["badges"][n] != undefined) {
+													if (settings[i][type]["values"]["badges"][n].hasOwnProperty(types[m])){
+														values["rgShowcaseConfig[5][" + n + "][" + types[m] + "]"] = settings[i][type]["values"]["badges"][n][types[m]] || values["rgShowcaseConfig[5][" + n + "][" + types[m] + "]"] || "";
+													}
+												}
+											}
+										}
+									}
+								}
+
+								break;
+
+							case 'rareachievements':
+								values["profile_showcase[1]"] = 1;
```

Comment:
That's why I used the Array first, until I did some further testing and found out that the index of that doesn't matter, it has no influence at all it seems. The only important thing is the order of the parameters in the request, and because I remove all `profile_showcase[]` if the showcases are changed and add them again to the JSON object and JSON is sorted by timestamp, it means the showcases still have the same order on the profile as in the array. If the user is level 10 and has 1 showcase slot and submits 2 showcases in the array he will get an error and the whole EditProfile Call is invalid (but changes to f.e. the items in an item showcase are still valid).